// etch/dist/default-scheduler.js
"use strict";

var _classCallCheck = function (instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } };

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

Object.defineProperty(exports, "__esModule", {
  value: true
});
// If the scheduler is not customized via `etch.setScheduler`, an instance of
// this class will be used to schedule updates to the document. The
// `updateDocument` method accepts functions to be run at some point in the
// future, then runs them on the next animation frame.

var DefaultScheduler = (function () {
  function DefaultScheduler() {
    _classCallCheck(this, DefaultScheduler);

    this.updateRequests = [];
    this.pendingAnimationFrame = null;
    this.performUpdates = this.performUpdates.bind(this);
  }

  _createClass(DefaultScheduler, [{
    key: "updateDocument",

    // Enqueues functions that write to the DOM to be performed on the next
    // animation frame. Functions passed to this method should *never* read from
    // the DOM, because that could cause synchronous reflows.
    value: function updateDocument(fn) {
      this.updateRequests.push(fn);
      if (!this.pendingAnimationFrame) {
        this.pendingAnimationFrame = window.requestAnimationFrame(this.performUpdates);
      }
    }
  }, {
    key: "getNextUpdatePromise",

    // Returns a promise that will resolve at the end of the next update cycle,
    // after all the functions passed to `updateDocument` and `updateDocumentSync`
    // have been run.
    value: function getNextUpdatePromise() {
      var _this = this;

      if (!this.nextUpdatePromise) {
        this.nextUpdatePromise = new Promise(function (resolve) {
          _this.resolveNextUpdatePromise = resolve;
        });
      }
      return this.nextUpdatePromise;
    }
  }, {
    key: "performUpdates",

    // Performs all the pending document updates. If running these update
    // functions causes *more* updates to be enqueued, they are run synchronously
    // in this update cycle without waiting for another frame.
    value: function performUpdates() {
      while (this.updateRequests.length > 0) {
        this.updateRequests.shift()();
      }

      // We don't clear the pending frame until the request queue is fully
      this.pendingAnimationFrame = null;

      if (this.nextUpdatePromise) {
        var resolveNextUpdatePromise = this.resolveNextUpdatePromise;
        this.nextUpdatePromise = null;
        this.resolveNextUpdatePromise = null;
        resolveNextUpdatePromise();
      }
    }
  }]);

  return DefaultScheduler;
})();

exports["default"] = DefaultScheduler;
module.exports = exports["default"];
